src/openeo_aggregator/backend.py

```diff
@@ -160,18 +172,58 @@ def _merge_collection_metadata(self, by_backend: Dict[str, dict]) -> dict:
                             or [[None, None]],
             },
         }
-        result["links"] = list(getter.union("links"))
+        # links
+        result["links"] = [l for l in list(getter.union("links")) if l["rel"] not in ("self", "parent", "root")]
+        result["links"].append({
+            "href": url_for("openeo.collections", _external=True),
+            "rel": "root"
+        })
+        result["links"].append({
+            "href": url_for("openeo.collections", _external=True),
+            "rel": "parent"
+        })
+        result["links"].append({
+            "href": url_for("openeo.collection_by_id", collection_id=result["id"], _external=True),
+            "rel": "self"
+        })
+        # cube_dimensions
         # TODO: combine cube:dimensions smarter?
         cube_dimensions = getter.first("cube:dimensions")
         if cube_dimensions:
             result["cube:dimensions"] = cube_dimensions
-        # TODO merge existing summaries?
-        result["summaries"] = {
-            # TODO: use a more robust/user friendly backend pointer than backend id (which is internal implementation detail)
-            self.STAC_PROPERTY_PROVIDER_BACKEND: list(by_backend.keys())
-        }
-        # TODO: assets ?
-
+        # summaries
+        # TODO: Better merging for summaries?
+        for summary in getter.get("summaries"):
+            print(summary)
+        result["summaries"] = {}
+        # TODO: use a more robust/user friendly backend pointer than backend id (which is internal implementation detail)
+        result["summaries"][self.STAC_PROPERTY_PROVIDER_BACKEND] = list(by_backend.keys())
+        # assets
+        # TODO: assets ? For now, take the union.
+        result["assets"] = list(getter.union("assets"))
+
+        # crs
+        crs_list = getter.union("crs", skip_duplicates=True)
+        if crs_list:
+            result["crs"] = list(crs_list)
+        # datasource_type
```

Comment:
what is this `data_source` field? Isn't that an internal VITO field?

Reply:
Excluded this field from the aggregator collection metadata!